backend/server/acl.go

```diff
@@ -340,13 +382,56 @@ func aclMiddleware(s *Server, pathPrefix string, ce *casbin.Enforcer, next echo.
 			}
 		}
 
+		if err := enforceIssueRouteACL(path, method, c.QueryParams(), principalID, role, getIsWorkspaceDeveloperMemberOfIssue(ctx, s.store)); err != nil {
+			return err
+		}
+
 		// Stores role into context.
 		c.Set(getRoleContextKey(), role)
 
 		return next(c)
 	}
 }
 
+// getIsWorkspaceDeveloperMemberOfIssue returns a function that checks if a principal is a member of an issue.
+// If the principal is one of the following, it is considered as a member of the issue:
+// 1. The creator of the issue.
+// 2. The member of the project that the issue belongs to.
+func getIsWorkspaceDeveloperMemberOfIssue(ctx context.Context, s *store.Store) func(issueID int, principalID int) error {
+	return func(issueID, principalID int) error {
+		issue, err := s.GetIssueV2(ctx, &store.FindIssueMessage{
+			UID: &issueID,
+		})
+		if err != nil {
+			return err
+		}
+		if issue == nil {
+			return errors.Errorf("cannot find issue %d", issueID)
+		}
+		if issue.Assignee.ID == principalID || issue.Creator.ID == principalID {
```

Comment:
"issue.Assignee.ID == principalID" this part is not mentioned in the comment

Reply:
Remove it.